agent/nmap_agent.py

```diff
@@ -63,6 +82,10 @@ def process(self, message: msg.Message) -> None:
         host = message.data.get("host", "")
         hosts: List[Tuple[str, int]] = []
 
+        # Connect to VPN
+        if self._vpn_config is not None:
+            self._connect_to_vpn()
```

Comment:
You need this function to be run once, so maybe even in the init. Also, the comment is not really useful since function name tells us what it does.